src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -396,7 +396,7 @@ private static IncrementalMethodStubGenerationContext CalculateStubInformation(M
                 locations,
                 callConv.ToSequenceEqualImmutableArray(SyntaxEquivalentComparer.Instance),
                 virtualMethodIndexData,
-                new ComExceptionMarshalling(),
+                new ComExceptionMarshalling(exceptionToUnmanagedMarshallerQualifiedName),
```

Comment:
Instead of editing `ComExceptionMarshalling`, please use the same logic as `ExceptionMarshalling.Custom`.

`ComExceptionMarshalling` specifically represents the default behavior for COM. Custom exception marshallers should replace that behavior with a different marshaller, not try to hook into it.

Reply:
Finished. I've kept the accessible-by-file-modifier logic in here as I'm not sure if that is checked by `CustomMarshallingInfoHelper.CreateNativeMarshallingInfoForNonSignatureElement` or not. If it is then we could cut down on the added resources as well.

Reply:
I don't believe that's already checked there, so lets keep the diagnostic.